src/script/lua_api/l_object.cpp

```diff
@@ -834,6 +837,86 @@ int ObjectRef::l_get_properties(lua_State *L)
 	return 1;
 }
 
+// set_observers(self, observers)
+int ObjectRef::l_set_observers(lua_State *L)
+{
+	GET_ENV_PTR;
+	ObjectRef *ref = checkObject<ObjectRef>(L, 1);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == nullptr)
+		throw LuaError("Invalid ObjectRef");
+
+	// Reset object to "unmanaged" (sent to everyone)?
+	if (lua_isnoneornil(L, 2)) {
+		sao->m_observers.reset();
+		return 0;
+	}
+
+	std::unordered_set<std::string> observer_names;
+	lua_pushnil(L);
+	while (lua_next(L, 2) != 0) {
+		std::string name = readParam<std::string>(L, -2);
+		if (name.empty())
+			throw LuaError("Observer name is empty");
+		if (name.size() > PLAYERNAME_SIZE)
+			throw LuaError("Observer name is too long");
+		if (!string_allowed(name, PLAYERNAME_ALLOWED_CHARS))
+			throw LuaError("Observer name contains invalid characters");
+		observer_names.insert(std::move(name));
+		if (!lua_toboolean(L, -1)) // falsy value?
+			throw LuaError("Values in the `observers` table need to be true");
+		lua_pop(L, 1); // pop value, keep key
+	}
+
+	RemotePlayer *player = getplayer(ref);
+	if (player != nullptr) {
+		if (observer_names.find(player->getName()) == observer_names.end())
+			throw LuaError("Players need to observe themselves");
+	}
+
+	sao->m_observers = observer_names;
+	return 0;
+}
+
+template<typename F>
+static int get_observers(lua_State *L, F get_observers)
```

Comment:
Renamed to `observer_getter`.

I'm not quite sure what you mean by your edit. Do you mean `F` -> `std::function<const Observers &(ServerActiveObject *)>`?

If so, I think a template makes sense here because we don't need any of the fancier uses of `std::function` (and the associated overhead, not that it would matter much);
we just want essentially the same function, but once accessing `m_observers` and once getting the effective observers.